services/web/server/src/simcore_service_webserver/folders/_folders_api.py

```diff
@@ -71,32 +92,36 @@ async def get_folder(
     folder_id: FolderID,
     product_name: ProductName,
 ) -> FolderGet:
-    user = await get_user(app, user_id=user_id)
+    folder_db = await folders_db.get_folder_db(
+        app, folder_id=folder_id, product_name=product_name
+    )
 
-    engine: Engine = app[APP_DB_ENGINE_KEY]
-    async with engine.acquire() as connection:
-        # NOTE: folder permissions are checked inside the function
-        folder_db: folders_db.FolderEntry = await folders_db.folder_get(
-            connection,
+    _private_workspace_user_id: UserID | None = user_id
+    if folder_db.workspace_id:
+        # Check access to workspace
+        await get_workspace(
+            app,
+            user_id=user_id,
+            workspace_id=folder_db.workspace_id,
             product_name=product_name,
-            folder_id=folder_id,
-            gids={user["primary_gid"]},
         )
+        # Setup folder user id to None, as this is not a private workspace
+        _private_workspace_user_id = None
+
```

Comment:
to make this easier to read, could you not do something like
```python
_private_workspace_user_id = try_get_workspace_user_id(...)
```

Reply:
could be re-used above as well

Reply:
ok I tried to improve it, let me know what you think.